### PR TITLE
Add support for Thread ID in Filebeat Kafka module

### LINKED ISSUE #18164: Kafka module support kafka logs from apache kafka 2.2.2 or higher version
**Describe the enhancement:**
As a user of Filebeat I would like to collect logs from Confluent kafka 5.2.3 (apache kafka 2.2.2). 

**Describe a specific use case for the enhancement or feature:**
Support logs coming from kafka 2.2.2

Sample logs: 
[state-change.log.gz](https://github.com/elastic/beats/files/4620510/state-change.log.gz)

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR adds support for Kafka 2.2.2 state change logs. A new field is introduced named `kafka.log.thread` which identifies the thread the log is coming from.

## Why is it important?

Previously, it was not supported. Now it is.

## Checklist

~~- [ ] My code follows the style guidelines of this project~~
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

Closes elastic/beats#18164

### CODE REVIEW

File: filebeat/module/kafka/log/ingest/pipeline.yml [hunk 1]
```diff
@@ -4,8 +4,9 @@ processors:
     field: message
     trace_match: true
     patterns:
-    - (?m)%{TIMESTAMP_ISO8601:kafka.log.timestamp}. %{LOGLEVEL:log.level} +%{JAVALOGMESSAGE:message}
+    - (?m)%{TIMESTAMP_ISO8601:kafka.log.timestamp}. ?%{LOGLEVEL:log.level} +%{JAVALOGMESSAGE:message}
```
Comment (ycombinator) [inline filebeat/module/kafka/log/ingest/pipeline.yml:7]: What is the purpose of the extra `?` before `%{LOGLEVEL:log.level}`?

File: filebeat/module/kafka/log/test/state-change-2.2.2.log-expected.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+[
```
Comment (ycombinator) [inline filebeat/module/kafka/log/test/state-change-2.2.2.log-expected.json:1]: Shouldn't there be a `state-change-2.2.2.log` file as part of this PR?

File: filebeat/module/kafka/log/ingest/pipeline.yml (hunk 1 above)
Comment (kvch) [reply filebeat/module/kafka/log/ingest/pipeline.yml]: Nothing. This PR was still in progress when you reviewed it. :P :D

Comment (ycombinator) [reply filebeat/module/kafka/log/ingest/pipeline.yml]: Sorry, @andresrc asked for some eyes on it so I reviewed it.

**Pull request merged**